tests/test_masked.py

```diff
@@ -254,3 +254,21 @@ def test_broadcasting(arrays):
         # NumPy simply drops the masked array.
         # We do not want to do the same quite intentionally.
         np.testing.assert_equal(a.to_numpy(), e)
+
+
+@pytest.mark.parametrize(
+    "np_array",
+    [
+        np.ma.masked_array([1, 2, 3], mask=[0, 0, 1], dtype=np.int64),
+        np.ma.masked_array([1, 2, 3], mask=[0, 0, 1], dtype=np.float64),
+        np.ma.masked_array([1, 2]),
+        np.ma.masked_array(["a", "b"], mask=True),
+        np.ma.masked_array([1, 2, 3], mask=[[[1]]]),
```

Comment:
```suggestion
        np.ma.masked_array([1, 2, 3], mask=[0, 0, 1], dtype=np.int64),
        np.ma.masked_array([1, 2, 3], mask=[0, 0, 1], dtype=np.float64),
        np.ma.masked_array([1.0, 2.0, 3.0], mask=[0, 0, 1]),
        np.ma.masked_array([1, 2]),
        np.ma.masked_array(["a", "b"], mask=True),
        np.ma.masked_array([1, 2, 3], mask=[[[1]]]),
        np.ma.masked_array([[1, 2, 3]], mask=[True, False, True]),
```
nit: can we add those tests to make sure that some more complex interactions are correctly covered?